integer/src/lib.rs

```diff
@@ -665,6 +665,47 @@ impl_integer_for_usize!(u32, test_integer_u32);
 impl_integer_for_usize!(u64, test_integer_u64);
 impl_integer_for_usize!(usize, test_integer_usize);
 
+/// An iterator over binomial coefficients.
+pub struct BinomialCoeff<T> {
+    a: T,
+    n: T,
+    k: T,
+}
+
+impl<T> BinomialCoeff<T>
+    where T: Integer,
+{
+    /// For a given n, iterate over all binomial coefficients ((k, n - k), binomial(n, k)).
```

Comment:
They are included for convenience. For example, when calculating `(x + y)**n`, the iterator would return the coefficients of `x**k * y**(n - k)`. I could live without them though, they are more important for the multinomial coefficients, where the order is not as obvious.